### PR TITLE
Finish default testing setup for new apps

### PR DESCRIPTION
**Pull request by flybayer**
### What are the changes and their implications?

This builds on the work started in #810 

- [x] Add top level `test` folder with `utils.ts` which includes custom `render` and `renderHook` test functions which include a router mock
- [x] Add `@testing-libary/react-hooks`
- [x] Clean up and improve `jest.config.js`
- [x] Move `app/pages/__tests__/index.test.ts` to `app/pages/index.test.ts` (co-location)
- [x] `skip` the default test because every person is going to break the test as soon as the replace the default content which is really annoying.
- [x] Add typechecking to test files with `ts-jest`
- [x] Parse test files with babel so jsx works
- [x] Make TS absolute paths and path aliases work in jest

### Checklist

- ~[ ] Tests added for changes~
- ~[ ] PR submitted to [blitzjs.com](https://github.com/blitz-js/blitzjs.com) for any user facing changes~

### CODE REVIEW

File: packages/core/src/with-router.tsx [hunk 1]
```diff
@@ -4,6 +4,8 @@ import {WithRouterProps as WithNextRouterProps} from "next/dist/client/with-rout
 import {useParams, extractRouterParams} from "./use-params"
 import {useRouterQuery} from "./use-router-query"
 
+export {RouterContext} from "next/dist/next-server/lib/router-context"
```
Comment (aem) [inline packages/core/src/with-router.tsx:7]: is this something we should write a test for in case this file moves?

File: packages/generator/templates/app/test/setup.ts [hunk 2]
```diff
@@ -2,4 +2,4 @@
 // allows you to do things like:
 // expect(element).toHaveTextContent(/react/i)
 // learn more: https://github.com/testing-library/jest-dom
-import '@testing-library/jest-dom/extend-expect';
+import "@testing-library/jest-dom/extend-expect"
```
Comment (peaonunes) [inline packages/generator/templates/app/test/setup.ts:5]: **question**: Do we have a prettier config for the template files? It's funny to see a lot of changes in the style on this PR. I am not sure what should be the one to follow.

File: packages/generator/templates/app/jest.config.js [hunk 3]
```diff
@@ -1,37 +1,29 @@
...
-    "app/**/*.{js,jsx,ts,tsx}",
-    "!app/**/*.d.ts"
-  ],
-  testMatch: [
-    "<rootDir>/app/**/__tests__/**/*.{js,jsx,ts,tsx}",
-    "<rootDir>/app/**/*.{spec,test}.{js,jsx,ts,tsx}"
-  ],
-  testEnvironment: "jest-environment-jsdom-fourteen",
-  modulePaths: [],
-  moduleFileExtensions: [
```
Comment (peaonunes) [inline packages/generator/templates/app/jest.config.js:21]: **question**: Why did you remove the file extensions to watch? Is that being replace by any combination of the rules you have added or we will track any file extension?

File: packages/generator/templates/app/test/utils.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
+import React from "react"
```
Comment (peaonunes) [inline packages/generator/templates/app/test/utils.tsx:1]: These utils are really nice 😄

File: packages/core/src/with-router.tsx (hunk 1 above)
Comment (flybayer) [reply packages/core/src/with-router.tsx:7]: The build will fail if it moves

File: packages/generator/templates/app/test/setup.ts (hunk 2 above)
Comment (flybayer) [reply packages/generator/templates/app/test/setup.ts:5]: It should be using the monorepo config. Not sure how the wrong format got in originally.

File: packages/generator/templates/app/jest.config.js (hunk 3 above)
Comment (flybayer) [reply packages/generator/templates/app/jest.config.js:21]: The default includes js, jsx, ts, tsx, and node. And the extra entries with `web` are not needed from what I can tell.

Comment (peaonunes) [reply packages/generator/templates/app/jest.config.js:21]: Awesome!

File: packages/core/src/with-router.tsx (hunk 1 above)
Comment (aem) [reply packages/core/src/with-router.tsx:7]: 🤦‍♂️ it's been a long day
